xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -1849,32 +1951,32 @@ struct AppModel: ModelProtocol {
         return Update(state: state, fx: fx)
     }
     
-    static func requestProvisionGateway(
+    static func requestRedeemInviteCode(
         state: AppModel,
         environment: AppEnvironment,
         inviteCode: InviteCode
     ) -> Update<AppModel> {
         guard let did = state.sphereIdentity,
               let did = Did(did) else {
+            logger.error("Missing identity, cannot redeem invite code")
```

Comment:
Is there a recovery path for this case?

Reply:
We should only get here if `.createSphere` has failed or wasn't run in the first place. 

We _could_ dispatch `.createSphere` here as an escape hatch and then retry... but we might end up in an infinite loop. I've slightly tweaked it to surface a UI error and try to create the sphere (once) if this happens.